### PR TITLE
Adding unit testing

### PR DESCRIPTION
**Pull request by dannyreilman**
Basic unit testing example

### CODE REVIEW

File: tests/bot_test.py [hunk 1]
```diff
@@ -1,17 +1,50 @@
...
+        ctx = create_mock_ctx(mockSend)
+        await bot.bing(ctx)
+        mockSend.assert_called_once_with("bong")
+
+    async def test_ping(self):
+        mockSend = mock.MagicMock()
+        ctx = create_mock_ctx(mockSend)
+        await bot.ping(ctx)
+        mockSend.assert_called_once_with("pong")
+
```
Comment (BDokas) [inline tests/bot_test.py:30]: Why isn't there a test_ding method?

**Review by BDokas — APPROVED** [review]

File: tests/test_helpers.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+from unittest import mock
+import datetime
+
+import pytest
+
+class MockCtx(mock.Mock):
+    @pytest.mark.asyncio
```
Comment (gobisa) [inline tests/test_helpers.py]: what if we put the functionality of `create_mock_ctx` in an `__init__(self, mockSend)` here

Comment (dannyreilman) [reply tests/test_helpers.py]: Ok now this suggestion is what I'm talking about!

File: tests/test_helpers.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+    def __init__(self, mockSend):
+        super().__init__()
+
+        def make_coroutine(mock):
+            async def coroutine(*args, **kwargs):
+                return mock(*args, **kwargs)
+            return coroutine
+
+        self.send = make_coroutine(mockSend)
+
```
Comment (gobisa) [inline tests/test_helpers.py:13]: beautiful

**Review by gobisa — APPROVED** [review]

File: tests/bot_test.py [hunk 4]
```diff
@@ -1,17 +1,45 @@
...
+    
+    async def test_day(self):
+        with mock.patch('bot.datetime') as mock_date:
+            mock_date.today.return_value = dt.datetime(2021, 8, 30)
+            mockSend = mock.Mock()
+            ctx = MockCtx(mockSend)
+            await bot.day(ctx)
+            mockSend.assert_called_once_with("It is Monday")
+
+    async def test_add(self):
```
Comment (BDokas) [inline tests/bot_test.py:39]: we should change the name of this method to be more accurate then

**Pull request merged**